### PR TITLE
[Python][DateTime] Add new Specs cases on DateTime

### PR DESCRIPTION
**Pull request by Maihav**
### Description

This PR adds new Spec cases on **recognizers-date-time** library, after the implementation of Matcher on Datetime. In order to make those tests pass, we reviewed the code and polished some bits of the implementation based on the .NET solution.

### Changes made

	• We enabled Specs previously marked as NotSupported in Python. As a result, we have **934‬** new tests giving a result of **11639** tests; which **8566** are passing successfully and the rest are skipped.
	
	• We added properties and regexes not initialized on the following files:
		○ date_extractor_config.py
		○ dateperiod_extractor_config.py
		○ datetime_extractor_config.py
		○ datetimeperiod_extractor_config.py
		○ timeperiod_extractor_config.py
		○ time_extractor_config.py
		○ set_extractor_config.py
		○ merged_extractor_config.py
		○ holiday_extractor_config.py
		○ duration_extractor_config.py
	
	• We implemented missing methods in:
		○ ChineseDatePeriodExtractor
		○ ChineseMergedExtractor		
		
	• We updated the following files:
		○ utilities.py: applying minor fixes on match_end() method
		○ base_duration.py: updating the tag_inequality_prefix() method
		○ base_datetimeperiod.py: adding minor changes on merge_date_with_time_period_suffix() method
		○ base_datetimeperiod.py: updating merge_two_time_points() method
		○ base_datetimeperiod.py: updating match_duration() method
		○ base_dateperiod.py: adding missing implementation on match_duration() method
		○ base_dateperiod.py: update match_simple_cases() method
		○ base_dateperiod.py: applying minor fixes on match_year_period() method
		○ base_timeperiod.py: updating merge_two_time_points() method
		○ base_timeperiod.py: changing has_connector_token by is_connector_token property
		○ base_timeperiod.py : adding minor changes
	
	• We standardized test case input following the same approach as in MS[#953](https://github.com/microsoft/Recognizers-Text/pull/953)

### Testing
Below you can find both screenshots, before and after, of the running tests passing successfully. You can notice the new Specs added.
image

### CODE REVIEW

**Comment by tellarin** [comment]
@Maihav we'll start reviewing. Meanwhile, can you folks resolve the merge conflicts?
Great work! Thanks!

File: Specs/DateTime/Turkish/DateTimePeriodExtractor.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [
   {
     "Input": "Bugün saat beşten yediye kadar yokum",
+    "NotSupported": "dotnet",
```
Comment (tellarin) [inline Specs/DateTime/Turkish/DateTimePeriodExtractor.json]: These cases are currently supported. Why adding this tag?

File: Specs/DateTime/Chinese/DateTimeModel.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "Context": {
       "ReferenceDateTime": "2018-01-22T00:00:00"
     },
-    "NotSupportedByDesign": "Java",
+    "NotSupportedByDesign": "Java, python",
```
Comment (tellarin) [inline Specs/DateTime/Chinese/DateTimeModel.json]: Why mark as NotSupportedByDesign?

Comment (tellarin) [reply Specs/DateTime/Chinese/DateTimeModel.json]: Did it work previously?
If not, but Chinese implementation exists, tests can be marked as NotSupported to indicate this is temporary.
NotSupportedByDesign means that there is no implementation for this type in the target platform and no current plans to add it.

File: Specs/DateTime/Chinese/DateTimeModel.json [hunk 3]
```diff
@@ -3812,29 +3812,5 @@
         }
       }
     ]
-  },
-  {
-    "Input": "他5天前就回来了",
```
Comment (tellarin) [inline Specs/DateTime/Chinese/DateTimeModel.json]: Why delete this spec?

File: Specs/DateTime/English/DateTimeParser.json [hunk 4]
```diff
@@ -1274,6 +1274,7 @@
     "Context": {
       "ReferenceDateTime": "2018-11-21T12:00:00"
     },
+    "NotSupported": "python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeParser.json]: Did this case work previously? Why is it no longer supported?

File: Specs/DateTime/Turkish/DateTimePeriodParser.json [hunk 5]
```diff
@@ -4,6 +4,7 @@
     "Context": {
       "ReferenceDateTime": "2016-11-07T16:12:00"
     },
+    "NotSupported": "dotnet",
```
Comment (tellarin) [inline Specs/DateTime/Turkish/DateTimePeriodParser.json]: Same.

**Comment by Aliandi** [comment]
Hey @tellarin, we had some old specs on our branch, we've updated with what's now on master. Sorry!

File: Specs/DateTime/English/DateTimeModel.json [hunk 6]
```diff
@@ -330,6 +330,7 @@
     "Context": {
       "ReferenceDateTime": "2016-11-07T00:00:00"
     },
+    "NotSupported": "python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Were all these new "NotSupported": "python" supported in Python previously, but do not work now?

File: Specs/DateTime/Spanish/DatePeriodParser.json [hunk 7]
```diff
@@ -935,6 +935,7 @@
     "Context": {
       "ReferenceDateTime": "2016-11-07T00:00:00"
     },
+    "NotSupported": "python",
```
Comment (tellarin) [inline Specs/DateTime/Spanish/DatePeriodParser.json]: Similar question for these Spanish cases.

**Comment by Aliandi** [comment]
@tellarin feedback applied!

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/english/merged_parser_config.py [hunk 8]
```diff
@@ -66,6 +84,13 @@ def set_parser(self) -> BaseSetParser:
         return self.__set_parser
 
     def __init__(self, config):
+        self._equal_regex = RegExpUtility.get_safe_reg_exp(BaseDateTime.EqualRegex)
+        self._suffix_after = RegExpUtility.get_safe_reg_exp(
```
[the remark this answers is not in the document]
Comment (tellarin) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/english/merged_parser_config.py]: I meant not to break such lines. But this is fine.

File: Specs/DateTime/English/DateTimeModel.json [hunk 9]
```diff
@@ -4889,6 +4875,7 @@
     "Context": {
       "ReferenceDateTime": "2018-07-02T00:00:00"
     },
+    "NotSupported": "python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: In next PRs, when marking an existing test with NotSupported in a PR, please add a Comment attribute describing the reason, ok?

File: Specs/DateTime/English/DateTimeModel.json [hunk 10]
```diff
@@ -9700,15 +9679,15 @@
     "Context": {
       "ReferenceDateTime": "2018-12-13T12:00:00"
     },
-    "NotSupportedByDesign": "java",
+    "NotSupportedByDesign": "java, python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Same

File: Specs/DateTime/English/DateTimeModel.json [hunk 11]
```diff
@@ -11758,6 +11733,7 @@
     "Context": {
       "ReferenceDateTime": "2019-05-30T12:00:00"
     },
+    "NotSupported": "python",
     "Results": [
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Same

File: Specs/DateTime/English/DateTimeModel.json [hunk 12]
```diff
@@ -12884,7 +12860,7 @@
     "Context": {
       "ReferenceDateTime": "2019-07-04T00:00:00"
     },
-    "NotSupported": "java",
+    "NotSupported": "java, python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Same

File: Specs/DateTime/English/DateTimeModel.json [hunk 13]
```diff
@@ -12915,7 +12891,7 @@
...
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Same

File: Specs/DateTime/English/DateTimeModel.json [hunk 14]
```diff
@@ -13459,6 +13435,7 @@
     "Context": {
       "ReferenceDateTime": "2019-07-17T00:00:00"
     },
+    "NotSupported": "python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Same

File: Specs/DateTime/English/DateTimeModel.json [hunk 15]
```diff
@@ -14365,15 +14342,15 @@
...
+    "NotSupported": "javascript",
     "Results": []
   },
   {
     "Input": "It happened due to a one second time difference.",
     "Context": {
       "ReferenceDateTime": "2019-09-09T00:00:00"
     },
-    "NotSupported": "java",
+    "NotSupported": "java, python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Same

**Pull request merged**